cranelift-codegen/meta/src/cdsl/regs.rs

```diff
@@ -364,19 +365,21 @@ impl IsaRegsBuilder {
             }
         }
 
-        // This limit should be coordinated with the `RegClassMask` and `RegClassIndex` types in
-        // isa/registers.rs of the non-meta code.
-        assert!(self.classes.len() <= 32, "Too many register classes");
+        assert!(
+            self.classes.len() <= constants::MAX_NUM_REG_CLASSES,
+            "Too many register classes"
+        );
 
-        // The maximum number of top-level register classes which have pressure tracking should be
-        // kept in sync with the MAX_TRACKED_TOPRCS constant in isa/registers.rs of the non-meta
-        // code.
         let num_toplevel = self
             .classes
             .values()
             .filter(|x| x.toprc == x.index && self.banks.get(x.bank).unwrap().pressure_tracking)
             .count();
-        assert!(num_toplevel <= 4, "Too many top-level register classes");
+
+        assert!(
+            num_toplevel <= constants::MAX_TRACKED_TOPRCS,
```

Comment:
I read "TOPRCS" as "topics"; maybe "MAX_TRACKED_TOP_REG_CLASSES"?

Reply:
Or `MAX_TRACKED_TOP_RCS` if you want to keep it shortish.